assets/js/googlesitekit/widgets/datastore/widgets.js

```diff
@@ -220,28 +210,22 @@ export const selectors = {
 
 		const { areaAssignments, widgets } = state;
 
-		const registryKey = select( STORE_NAME ).getWidgetRegistryKey();
-
-		return Object.values( widgets ).filter( ( widget ) => {
-			return areaAssignments[ widgetAreaSlug ] && areaAssignments[ widgetAreaSlug ].includes( widget.slug );
-		} ).sort( ( widgetA, widgetB ) => {
-			if ( widgetA.priority > widgetB.priority ) {
-				return 1;
-			}
-
-			if ( widgetA.priority < widgetB.priority ) {
-				return -1;
-			}
+		const registryKey = select( CORE_SITE ).getRegistryKey();
 
-			return 0;
-		} ).map( ( widget ) => {
-			const widgetWithComponent = { ...widget };
-			if ( WidgetComponents[ registryKey ] ) {
-				widgetWithComponent.component = WidgetComponents[ registryKey ][ widget.slug ];
-			}
+		const sorted = sortObjectMapByKey(
+			Object.values( widgets ).filter( ( widget ) => {
+				return areaAssignments[ widgetAreaSlug ] && areaAssignments[ widgetAreaSlug ].includes( widget.slug );
+			} ).map( ( widget ) => {
```

Comment:
See my above comment, here we're passing an array to `sortObjectMapByKey` which is not correct as it works now. However, I think we should simplify that function to deal with arrays, at which point this would become okay.